include/alpaka/atomic/AtomicUniformCudaHip.hpp

```diff
@@ -0,0 +1,474 @@
+/* Copyright 2022 René Widera
+ *
+ * This file is part of alpaka.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+#pragma once
+
+#if defined(ALPAKA_ACC_GPU_CUDA_ENABLED) || defined(ALPAKA_ACC_GPU_HIP_ENABLED)
+
+#    include <alpaka/atomic/Op.hpp>
+#    include <alpaka/core/BoostPredef.hpp>
+#    include <alpaka/core/Positioning.hpp>
+
+#    include <type_traits>
+
+namespace alpaka
+{
+    //! The GPU CUDA/HIP accelerator atomic ops.
+    //
+    //  Atomics can be used in the hierarchy level grids, blocks and threads.
+    //  Atomics are not guaranteed to be save between devices.
+    class AtomicUniformCudaHipBuiltIn
```

Comment:
Does it also works, if the CUDA and the HIP are enabled at the same time?

Reply:
As I know. You can not enable CUDA and HIP at the same time.

Reply:
Really, I thought Andrea implemented it? Or is it permitted by the CMake?

Reply:
As I know he is compiling a library for CUDA and a library for HIP and linking both together.
Both libs will use a different compiler.

@fwyzard Could you check if this PR is working for you. I do not exactly know your workflows.

Reply:
I think I remember that we introduced the device API struct to avoid naming conflicts. Nevertheless, these changes are fully on the device side and should not create symbols on the host side which can conflict.

Reply:
Sorry, I have been doing night shifts this week, and had little time for the rest :-/

I'll try to have a look ASAP.